### DISCUSSION TITLE
Invalid value for: local_ts

### DISCUSSION THREAD

**Discussion by KushalGowda-Visteon** [strongswan/strongswan#2547] [Q&A] [answered]
I am trying to establish secure connection between android(100.64.10.4) and QNX(100.64.10.3) of same machine.
Line causing error:
`local_ts = 100.64.10.4/32[udp/22000-22015]`
<img width="569" alt="image" src="https://github.com/user-attachments/assets/820e8228-438f-47a3-af7e-510103f8d54a">

 when I try to load the swanctl file. ./swanctl --load-all
Error:  Loading connection 'eva25-initiator' failed: **invalid value for: local_ts, config discarded** loaded 0 of 1 connections, 1 failed to load, 0 unloaded

while debugging, I removed port range in local_ts the config file loads successfully. So clearly the issue is with port range **[udp/22000-22015\]**. 
However the same configuration is working fine in QNX side with port range defined.
Is there any necessary plugins or any other changes that I should make to load swanctl.conf load successfully in android side with port range defined.

**Comment by tobiasbrunner** [comment]
I can't reproduce the error. What version are you using? Could you post the actual config?

Also, did you configure those duplicate child configs (`eva25-publisher/subscriber` having the same TS) on purpose? (Note that this won't work.)

**Reply by KushalGowda-Visteon** [reply]
I understand the issue cannot be reproduced on your setup. I'm using version 5.9.14. Here’s the actual configuration I’m using (even this doesn’t work for local_ts):
```
connections {  
    ubuntu-to-b {  
        local_addrs = 100.64.10.4  
        remote_addrs = 100.64.10.3  
        version = 2  
        proposals = aes128-sha256-prfsha256-ecp256  

        local {  
            auth = psk  
            id = 100.64.10.4  
        }  
        remote {  
            auth = psk  
            id = 100.64.10.3  
        }  

        children {  
            esp-transport {  
                mode = transport  
                esp_proposals = aes128gcm16-esn  
                local_ts = 100.64.10.4[udp/22000-22015]  
                remote_ts = 100.64.10.3[udp/22000-22015]  
            }  
        }  
    }  
}  

secrets {  
    ike {  
        id = 100.64.10.3  
        secret = 0x01010101010101010101010101010101  
    }  
}
```
Regarding the duplicate child configs, no, this wasn’t intentional.

**Reply by KushalGowda-Visteon** [reply]
Also added following modules expecting to resolve port range issue. No effect however.
CONFIG_NET_KEY=y
CONFIG_XFRM=y
CONFIG_XFRM_USER=y
CONFIG_XFRM_ALGO=y
CONFIG_INET=y
CONFIG_NETFILTER_XT_TARGET_TPROXY=y
CONFIG_NETFILTER_XT_MARK=y
CONFIG_IP_NF_IPTABLES=y
CONFIG_NETFILTER_XT_TARGET_LOG=y
CONFIG_NET=y
CONFIG_NETFILTER_NETLINK=y
CONFIG_XFRM_IPCOMP=y
CONFIG_INET_XFRM_MODE_UDP=y
CONFIG_CRYPTO_AES=y
CONFIG_CRYPTO_SHA256=y
CONFIG_CRYPTO_HMAC=y
CONFIG_CRYPTO_USER=y
CONFIG_INET=y
CONFIG_IPV6=y

**Reply by KushalGowda-Visteon** [reply]
We are using the below android architecture:
 file /vendor/bin/charon
/vendor/bin/charon: ELF shared object, 64-bit LSB arm64, dynamic (/system/bin/linker64), for Android 31, built by NDK r23b (7779620), not stripped

**loaded plugins**: charon aes des rc2 sha2 sha1 md5 mgf1 random nonce x509 revocation constraints pubkey pkcs1 pkcs7 pkcs12 pgp dnskey sshkey pem openssl pkcs8 fips-prf gmp curve25519 xcbc cmac hmac kdf gcm drbg attr kernel-pfkey kernel-netlink resolve socket-default socket-dynamic stroke vici updown eap-mschapv2 xauth-generic android-log whitelist led addrblock counters

Below zip file consists of 
1. strongswan configuration(strongswan.conf), 
2. swanctl configuration(swanctl.conf),
3. test logs,
4. charon and swanctl executables.
[Configuration_logs_executables.zip](https://github.com/user-attachments/files/18004947/Configuration_logs_executables.zip)

**Reply by tobiasbrunner** [reply, marked as answer]
> Here’s the actual configuration I’m using (even this doesn’t work for local_ts):

Works fine here, even with 5.9.14. What could be an issue is that you used `udp` in the traffic selector, which is resolved via `getprotobyname()`. That might not work on Android. So please try to use the numeric protocol identifier for UDP, which is 17.

**Reply by KushalGowda-Visteon** [reply]
Thank you so much. It worked.
[Android_working_case.txt](https://github.com/user-attachments/files/18018895/Android_working_case.txt)
